frame/support/src/traits/misc.rs

```diff
@@ -62,9 +63,20 @@ pub trait DefensiveUnwrap<T> {
 	fn defensive_unwrap_or_default(self) -> T
 	where
 		T: Default;
+
+	/// Same as [`defensive_map`], but it does not alter the inner value, but it will log warnings
+	/// if the inner value is `None` or `Err`.
+	///
+	/// This is useful as:
+	/// ```nocompile
+	/// if let Some(inner) = maybe_value().defensive() {
+	/// 	 	..
+	/// }
+	/// ```
+	fn defensive(self) -> Self;
```

Comment:
What made you name it `defensive` instead of `defensive_unwrap`?

Reply:
`unwrap` means you unpack `Option<T>` or `Result<T, _>` to `T` (e.g return type would have been `-> T`), here I am not unpacking the data, only logging errors if it is an error/none.

Reply:
we can add `defensive_unwrap()` as well, but it is not really aligned with how we code in the runtime and the whole business of defensive programming. We want to code to continue without panic even in the case of a super unforeseen `Err` or `None`. `defensive_unwrap` is not useful for this.

Reply:
Yes makes sense, other there is no value to pass on.